content/lessons/chapter-8/building-blocks-7.tsx

```diff
@@ -176,11 +174,11 @@ for bhash in candidates:
   const jsValidation = `const HEIGHT = 6929851
 const candidates = Bitcoin.rpc("getblocksbyheight", HEIGHT)
 for (const bhash of candidates) {
-let block = Bitcoin.rpc("getblock", bhash)
-if(validateBlock(block)){
-  return bhash;
-    }
-  }`
+  let block = Bitcoin.rpc("getblock", bhash)
+  if(validateBlock(block) == true){
```

Comment:
I think we can improve the validation even further here. We have access to `chainstate.json` in the backend which has the validity of each block in it already

Reply:
I haven't tested it yet but I think we will want something akin to this.
```
    program: `//BEGIN VALIDATION BLOCK
const fs = require('fs');
let HEIGHT = 6929851
let candidates = Bitcoin.rpc("getblocksbyheight", HEIGHT)

const state = JSON.parse(fs.readFileSync('chainstate.json', 'utf8'));

const isValid = JSON.stringify(state.valid) === JSON.stringify(answer.valid)
for (const bhash of candidates){
  let block = Bitcoin.rpc("getblock", bhash)

  if(validateBlock(block) == true && state.block.valid){
    console.log(bhash)
  }
}
console.log("KILL")`,
```

Reply:
I just tested this and if I set my solution for `validateBlock()` to just be `return true;`, I pass the challenge. The valid block check is good, but it's not checking the invalid blocks.

I'm going to change the challenge so the pre-loaded code includes the loop that tries all the block candidates and prints the hash of the valid one. Then we can just check that the user printed the correct block hash.

Reply:
On second thought, I think I should do that in a follow up PR. This challenge is currently a little buggy on prod and I'd prefer to push out a fix as soon as possible. What are your thoughts?

Reply:
I just rebased that last commit, please see eed9fa67fcb428843b544fc4b76e89912d6169bb for an update

In that commit I updated the validation code to make sure `validateBlock()` returns false for invalid blocks, and added a custom error message for when invalid blocks are getting through. Also updated the challenge instructions to tell the user to return a boolean. Now that the validation code is updated, the player no longer needs to print the hash of the valid block and the spoilers page does not need any updates.